app/models/room.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Room < ApplicationRecord
+  validates :room_hash, uniqueness: true, allow_nil: false
+  before_create :generate_room_hash
```

Comment:
I'd go with `after_initialize` and `||=` in the assignment (would make it easier in specs to have the public id set and allows the override if needed)

Reply:
With a change to `to_param` returning the room_hash, do you think this would be an issue, specifically that `to_param` would return the room_hash even though the record hasn't been saved to the database?

Reply:
there should be no use case where we don't save the record in the db (unless you check for the status of `.save` always use the bang methods to raise, ie `save!`/`.create!`